docs/testplan/DWRR-Snappi-Test-Plan.md

```diff
@@ -0,0 +1,797 @@
+
+#### 2.4.4 Test Procedure
+1. Identify DSCP-to-queue mapping on the switch for DSCP values 1, 2, 3, and 4
+2. Configure DWRR scheduling with 30:40:15:15 weight ratio for the four queues
+3. Configure traffic flows on the traffic generator:
+   - Flow 1: 35 Gbps with DSCP 1
+   - Flow 2: 45 Gbps with DSCP 2
+   - Flow 3: 25 Gbps with DSCP 3
+   - Flow 4: 25 Gbps with DSCP 4
+4. Start traffic and capture on the egress port
+5. After 60 seconds, stop traffic and capture
+6. Analyze captured traffic by filtering based on DSCP values
+
+#### 2.4.5 Expected Results
+- Queue 1 (DSCP 1) traffic should account for 30% of the egress traffic: 30 Gbps
+- Queue 2 (DSCP 2) traffic should account for 40% of the egress traffic: 40 Gbps
+- Queue 3 (DSCP 3) traffic should account for 15% of the egress traffic: 15 Gbps
+- Queue 4 (DSCP 4) traffic should account for 15% of the egress traffic: 15 Gbps
+
+#### 2.4.6 Success Criteria
+- The proportion of traffic from each queue must match the configured weights within a 2% tolerance
+  - Queue 1: 28-32% (28-32 Gbps)
+  - Queue 2: 38-42% (38-42 Gbps)
+  - Queue 3: 13-17% (13-17 Gbps)
+  - Queue 4: 13-17% (13-17 Gbps)
+
+### 2.5 Scenario 5: DWRR Validation with Different DSCP Distributions on Each Port
+
+#### 2.5.1 Test Objective
+Validate DWRR functionality with mixed DSCP traffic from various ingress ports to validate proper queuing regardless of traffic source.
+
+#### 2.5.2 Topology
+
+```mermaid
+flowchart LR
+    subgraph "Traffic Generator"
+        TX1[TX Port 1 100G Mixed DSCP]
+        TX2[TX Port 2 100G Mixed DSCP]
+    end
+
+    subgraph "Switch Under Test"
+        SW[DUT]
+        Q1[Queue 1 Weight: 25%]
+        Q2[Queue 2 Weight: 25%]
+        Q3[Queue 3 Weight: 25%]
+        Q4[Queue 4 Weight: 25%]
+    end
+
+    subgraph "Traffic Analyzer"
+        RX[RX Port 100G]
+    end
+
+    TX1 -->|DSCP 1,2,3,4| SW
+    TX2 -->|DSCP 1,2,3,4| SW
+    SW --> Q1
+    SW --> Q2
+    SW --> Q3
+    SW --> Q4
+    Q1 -->|Expected 25%| RX
+    Q2 -->|Expected 25%| RX
+    Q3 -->|Expected 25%| RX
+    Q4 -->|Expected 25%| RX
+
+    classDef congested fill:#f96,stroke:#333,stroke-width:2px;
+    class RX congested;
+```
+
+#### 2.5.3 Test Configuration
+- **Ingress Ports**: 2 logical ports (100G each)
+- **Egress Port**: 1 logical port (100G)
+- **Traffic Pattern**:
+  - Port 1: Mixed traffic with the following distribution:
+    * 20G with DSCP 1 (mapped to Queue 1)
+    * 25G with DSCP 2 (mapped to Queue 2)
+    * 15G with DSCP 3 (mapped to Queue 3)
+    * 20G with DSCP 4 (mapped to Queue 4)
+  - Port 2: Mixed traffic with the following distribution:
+    * 20G with DSCP 1 (mapped to Queue 1)
+    * 15G with DSCP 2 (mapped to Queue 2)
+    * 25G with DSCP 3 (mapped to Queue 3)
+    * 20G with DSCP 4 (mapped to Queue 4)
+  - Total ingress traffic: 160G (creates 60% congestion on 100G egress port)
+- **Traffic Flow Duration**: 60 seconds
+- **Packet Size**: Mixed packet sizes (64, 512, 1024, 1500 bytes)
```

Comment:
missing 4k, 8k here. 

1024 and 1500 doesn't seem to be diff much, but ok to add both.